adam-core/src/main/scala/org/bdgenomics/adam/rdd/sequence/SliceRDD.scala

```diff
@@ -0,0 +1,232 @@
+ *
+ * @param rdd An RDD of genomic Slices.
+ * @param sequences The reference genome this data is aligned to.
+ */
+case class SliceRDD(rdd: RDD[Slice],
+                    sequences: SequenceDictionary) extends AvroGenomicRDD[Slice, SliceRDD] with Logging {
+
+  protected def buildTree(rdd: RDD[(ReferenceRegion, Slice)])(
+    implicit tTag: ClassTag[Slice]): IntervalArray[ReferenceRegion, Slice] = {
+    IntervalArray(rdd, SliceArray.apply(_, _))
+  }
+
+  /**
+   * Save slices as Parquet or FASTA.
+   *
+   * If filename ends in .fa or .fasta, saves as FASTA. If not, saves slices
+   * to Parquet. Defaults to 60 character line length, if saving to FASTA.
+   *
+   * @param filePath Path to save files to.
+   * @param asSingleFile If true, saves output as a single file.
+   */
+  def save(filePath: java.lang.String, asSingleFile: java.lang.Boolean) {
+    if (filePath.endsWith(".fa") || filePath.endsWith(".fasta")) {
+      saveAsFasta(filePath, asSingleFile = asSingleFile)
+    } else {
+      if (asSingleFile) {
+        log.warn("asSingleFile = true ignored when saving as Parquet.")
+      }
+      saveAsParquet(new JavaSaveArgs(filePath))
+    }
+  }
+
+  /**
+   * Save slices in FASTA format.
+   *
+   * @param filePath Path to save files to.
+   * @param asSingleFile If true, saves output as a single file.
+   * @param lineWidth Hard wrap FASTA formatted slice at line width, default 60.
+   */
+  def saveAsFasta(filePath: String, asSingleFile: Boolean = false, lineWidth: Int = 60) {
+
+    def toFasta(slice: Slice): String = {
+      val sb = new StringBuilder()
+      sb.append(">")
+      sb.append(slice.getName)
+      Option(slice.getDescription).foreach(n => sb.append(" ").append(n))
+      sb.append(s" slice.getStart-slice.getEnd:slice.getStrand")
+      for (line <- Splitter.fixedLength(lineWidth).split(slice.getSequence)) {
+        sb.append("\n")
+        sb.append(line)
+      }
+      sb.toString
+    }
+
+    writeTextRdd(rdd.map(toFasta), filePath, asSingleFile)
+  }
+
+  /**
+   * Merge slices into sequences.
+   *
+   * @return Returns a SequenceRDD containing merged slices.
+   */
+  def merge(): SequenceRDD = {
+    def toSequence(slice: Slice): Sequence = {
+      Sequence.newBuilder()
+        .setName(slice.getName)
+        .setDescription(slice.getDescription)
+        .setAlphabet(slice.getAlphabet)
+        .setSequence(slice.getSequence)
+        .setLength(slice.getLength)
+        //.setAttributes(slice.getAttributes)
+        .build
+    }
+
+    def mergeSequences(first: Sequence, second: Sequence): Sequence = {
+      Sequence.newBuilder(first)
+        .setLength(first.getLength + second.getLength)
+        .setSequence(first.getSequence + second.getSequence)
+        .build
+    }
+
+    val merged: RDD[Sequence] = rdd
+      .sortBy(slice => (slice.getName, slice.getStart))
+      .map(slice => (slice.getName, toSequence(slice)))
+      .reduceByKey(mergeSequences)
```

Comment:
Is this different than that in [`NucleotideContigFragmentRDD`](https://github.com/bigdatagenomics/adam/blob/master/adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDD.scala#L208)?